### PR TITLE
Cleanup regex_expressions.rs to remove _regexp_match function 

### LINKED ISSUE #9106: Cleanup regex_expressions.rs to remove _regexp_match function
### Is your feature request related to a problem or challenge?

arrow-rs 50 includes this functionality mentioned in a comment in the file:
```
/// TODO: Remove this once it is included in arrow-rs new release.
/// <https://github.com/apache/arrow-rs/pull/5235>
fn _regexp_match<OffsetSize: OffsetSizeTrait>(
```
thus this function can be removed and the arrow-rs implementation can be called directly.

As noted in https://github.com/apache/arrow-datafusion/issues/9102#issuecomment-1922250473

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Omega359**
## Which issue does this PR close?

Closes #9106

## Rationale for this change
Code cleanup.

## What changes are included in this PR?

Code removal, minor code cleanup, toml file updates.

## Are these changes tested?

Yes

## Are there any user-facing changes?

No.

### CODE REVIEW

**Review by alamb — COMMENTED** [review]
Thanks @Omega359  -- the code looks great. I am running some benchmarks just to be sure

**Review by alamb — APPROVED** [review]
@viirya fixed the problem in https://github.com/apache/arrow-datafusion/pull/8631

I ran some of the queries from that PR:
```sql
❯ SELECT COUNT(*) FROM 'index' WHERE
    ARRAY_LENGTH(
      REGEXP_MATCH(path, '\\.(asm|c|cc|cpp|cxx|h|hpp|rs|[Ff][0-9]{0,2}(?:or)?|go)$')
    ) > 0;
```

on 7.2 GB of parquet data
```
$ du -s -h index
7.2G	index
```


# Main (without this PR):
```
1 row in set. Query took 4.820 seconds.
```

# This PR:
```
1 row in set. Query took 4.760 seconds.
```

So basically the same. 

Thank you @Omega359  -- very nice cleanup 🧹

File: datafusion/physical-expr/src/regex_expressions.rs [hunk 1]
```diff
@@ -25,8 +25,7 @@ use arrow::array::{
     new_null_array, Array, ArrayDataBuilder, ArrayRef, BufferBuilder, GenericStringArray,
     OffsetSizeTrait,
 };
-use arrow_array::builder::{GenericStringBuilder, ListBuilder};
-use arrow_schema::ArrowError;
+
```
Comment (viirya) [inline datafusion/physical-expr/src/regex_expressions.rs:28]:
```suggestion
```

File: datafusion/physical-expr/src/regex_expressions.rs [hunk 2]
```diff
@@ -409,10 +332,12 @@ pub fn specialize_regexp_replace<T: OffsetSizeTrait>(
 
 #[cfg(test)]
 mod tests {
-    use super::*;
     use arrow::array::*;
+
```
Comment (viirya) [inline datafusion/physical-expr/src/regex_expressions.rs:336]:
```suggestion
```

File: datafusion/physical-expr/src/regex_expressions.rs [hunk 3]
```diff
@@ -409,10 +332,12 @@ pub fn specialize_regexp_replace<T: OffsetSizeTrait>(
...
     use datafusion_common::ScalarValue;
 
+    use super::*;
```
Comment (viirya) [inline datafusion/physical-expr/src/regex_expressions.rs:339]:
```suggestion
    use datafusion_common::ScalarValue;
    use super::*;
```

**Review by viirya — APPROVED** [review]

**Comment by alamb** [comment]
Thank you @Omega359  and @viirya  🚀

**Pull request merged**